### PR TITLE
Add libedit/3.1 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libedit/3.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libedit/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        return "source_subfolder"
+
+    def requirements(self):
+        if self.options.terminal_db == "termcap":
+            self.requires("termcap/1.3.1")
+        elif self.options.terminal_db == "ncurses":
+            self.requires("ncurses/6.2")
+        elif self.options.terminal_db == "tinfo":
+            raise ConanInvalidConfiguration("tinfo is not (yet) available on CCI")
+            self.requires("libtinfo/x.y.z")
```
Comment (uilianries) [inline recipes/libedit/all/conanfile.py:39]:
```suggestion
```

File: recipes/libedit/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        if self.options.terminal_db == "termcap":
+            self.requires("termcap/1.3.1")
+        elif self.options.terminal_db == "ncurses":
+            self.requires("ncurses/6.2")
+        elif self.options.terminal_db == "tinfo":
```
Comment (uilianries) [inline recipes/libedit/all/conanfile.py:37]:
```suggestion
        elif self.options.terminal_db == "tinfo":
            # TODO - Add tinfo when available
```

**Comment by madebr** [comment]
> Shall we name this one `editline` and reserve `libedit` to the original one?
> 
> http://thrysoee.dk/editline/
> 
> > This is an autotool- and libtoolized port of the NetBSD Editline library (libedit). This Berkeley-style licensed command line editor library provides generic line editing, history, and tokenization functions, similar to those found in GNU Readline.

Good point.
I'll make the change.

**Comment by madebr** [comment]
@SSE4 @uilianries @jgsogo 
Please re-review!

**Pull request merged**